metrics-operator/controllers/common/providers/dynatrace/dynatrace_test.go

```diff
@@ -316,10 +325,11 @@ func TestEvaluateQuery_RefNotExistingKey(t *testing.T) {
 			TargetServer: svr.URL,
 		},
 	}
-
-	_, _, e := kdp.EvaluateQuery(context.TODO(), obj, p)
-	require.NotNil(t, e)
-	require.True(t, strings.Contains(e.Error(), "invalid key "+missingKey))
+	for _, obj := range objects {
+		_, _, e := kdp.EvaluateQuery(context.TODO(), obj, p)
+		require.NotNil(t, e)
+		require.True(t, strings.Contains(e.Error(), "invalid key "+missingKey))
+	}
 }
 
 func TestEvaluateQuery_HappyPath(t *testing.T) {
```

Comment:
Few lines afterward, in the callback of the "mocked" DT API response (line 336~), it would be good to verify that the timeframe was correctly set checking that a `from` query parameter is available. 

Something like:
```
p := r.URL.Query().Get("from")
require.NotEmpty(t, p)
```

Reply:
Added a new test case for the same which covers this.